src/v/kafka/client/direct_consumer/data_queue.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+#include "kafka/client/direct_consumer/api_types.h"
+
+namespace kafka::client {
+
+class data_queue {
```

Comment:
looks similar to https://github.com/redpanda-data/redpanda/blob/dev/src/v/transform/transfer_queue.h#L48


Consider adding light documentation